test/TestUtilities/Test.Utility/Signing/TrustedTestCert.cs

```diff
@@ -84,6 +85,14 @@ private void AddCertificateToStore()
             _store = new X509Store(StoreName, StoreLocation);
             _store.Open(OpenFlags.ReadWrite);
             _store.Add(TrustedCert);
+
+            //Add wait for Linux, as https://github.com/dotnet/runtime/issues/32608
+            //Windows has a live-synchronized model, and on Linux, there is a filesystem/rescan delay problem.
+            //For performance reasons, dotnet/runtime only check to see if the store has been modified once a second.
+            if (RuntimeEnvironmentHelper.IsLinux)
+            {
+                Thread.Sleep(500);
```

Comment:
First, the comment ("once a second") does not match the implementation (500 milliseconds).  If .NET Core's internal poll is 1 second, our window should be 2 seconds, minimum.

Second, in general I'd prefer polling over a hard-coded sleep; however, some consideration should be paid as to the impact of chain building results possibly being cached before a test actually runs.  (This whole discussion is limited to Linux, and I don't know what level of caching if any exists on Linux.)

Third, why are we not subject to the longer 30-second interval Jeremy mentions?  We load and modify the store within 1 second.  Here's Jeremy's relevant comment:
> The 1 second interval is how often it'll check for a new last modified time on the store. LMT values are only to the second, so if you end up causing the chain processor view of the store to load and modify the store again during the same second, you'll have to wait for the longer interval (30 seconds) before you actually see the benefit (that's the "just in case LMT is disabled for this filesystem, or we had a level-of-refinement problem, or clock adjustments" failsafe).

Reply:
Thanks! Changed to use polling and set more than 30 seconds for the upper limitation.